### PR TITLE
feat: local deployment cancellation

### PR DESCRIPTION
**Pull request by junfuchen99**
**Description of changes:**
1. Implement CLI command for local deployment cancellation - `deployment cancel -i deploymentId`
2. Implement server-side logics for handling cancellation.
3. Update SDK dependency.

**Why is this change necessary:**
To enable local deployment cancellation and bring parity with cloud deployment experience.

**How was this change tested:**
Added unit tests.

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: client/src/main/java/com/aws/greengrass/cli/adapter/impl/NucleusAdapterIpcClientImpl.java [hunk 1]
```diff
@@ -205,6 +207,23 @@ public String createLocalDeployment(CreateLocalDeploymentRequest createLocalDepl
...
 
+    @Override
+    public String cancelLocalDeployment(CancelLocalDeploymentRequest cancelLocalDeploymentRequest)  {
+        try {
+
+            CancelLocalDeploymentResponse cancelLocalDeploymentResponse =
+                    getIpcClient().cancelLocalDeployment(cancelLocalDeploymentRequest, Optional.empty())
+                            .getResponse().get(DEFAULT_TIMEOUT_IN_SEC, TimeUnit.SECONDS);
+            return cancelLocalDeploymentResponse.getMessage();
+        } catch (ExecutionException | TimeoutException | InterruptedException e) {
```
Comment (MikeDombo) [inline client/src/main/java/com/aws/greengrass/cli/adapter/impl/NucleusAdapterIpcClientImpl.java:218]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

**Remember**: Do this only for InterruptedException. Doing this for other types of exceptions may give rise to subtle bugs.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)

File: client/src/test/java/com/aws/greengrass/cli/commands/DeploymentCommandTest.java [hunk 2]
```diff
@@ -195,6 +196,22 @@ void GIVEN_WHEN_failure_handling_policy_provided_with_invalid_words_THEN_invalid
...
+        assertThat(exitCode, is(0));
+    }
+
+    @Test
+    void GIVEN_deployment_cancel_When_deploymentId_not_provided_THEN_request_sent() {
+        int exitCode = runCommandLine("deployment", "cancel");
+        verify(nucleusAdapteripc, never()).cancelLocalDeployment(any());
+        assertThat(exitCode, is(2));
+    }
+
```
Comment (Xuqing888) [inline client/src/test/java/com/aws/greengrass/cli/commands/DeploymentCommandTest.java:214]: Do we need a test like GIVEN_deployment_cancel_When_deploymentId_provided_isValid_THEN_xxxx

File: server/src/main/java/com/aws/greengrass/cli/CLIEventStreamAgent.java [hunk 3]
```diff
@@ -520,6 +529,99 @@ public void handleStreamEvent(EventStreamJsonMessage streamRequestEvent) {
...
+                    CancelLocalDeploymentResponse response = new CancelLocalDeploymentResponse();
+                    response.setMessage("Cancellation request is not processed because deployment is already finished");
+                    return response;
+                }
+
+                // reuse the previous deployment id and set isCancelled=true, because deployment queue replaces
+                // queued deployments if it has the same ids.
+                Deployment deployment = new Deployment(Deployment.DeploymentType.LOCAL, deploymentId, true);
+                if (deploymentQueue == null) {
+                    logger.atError().log("Deployments queue not initialized");
```
Comment (Xuqing888) [inline server/src/main/java/com/aws/greengrass/cli/CLIEventStreamAgent.java:583]: So we just error and do nothing? (is that queue initialization in scope of nucleus that CLI can't control?)

File: client/src/test/java/com/aws/greengrass/cli/commands/DeploymentCommandTest.java (hunk 2 above)
Comment (junfuchen99) [reply client/src/test/java/com/aws/greengrass/cli/commands/DeploymentCommandTest.java:214]: is that the same as `GIVEN_deployment_cancel_When_deploymentId_provided_THEN_request_sent`? The actual deployment id validation is done on server side so we can't test anything on client side more than if an argument is supplied or not.

File: server/src/main/java/com/aws/greengrass/cli/CLIEventStreamAgent.java (hunk 3 above)
Comment (junfuchen99) [reply server/src/main/java/com/aws/greengrass/cli/CLIEventStreamAgent.java:583]: Yes the queue is only managed by nucleus deployment service. CLI just offer a new deployment to the queue if possible

**Review by indougnito — APPROVED** [review]

File: server/src/main/java/com/aws/greengrass/cli/CLIEventStreamAgent.java (hunk 3 above)
Comment (junfuchen99) [reply server/src/main/java/com/aws/greengrass/cli/CLIEventStreamAgent.java]: updated the error message

**Review by Xuqing888 — APPROVED** [review]
